src/ar/xsanime/src/eu/kanade/tachiyomi/animeextension/ar/xsanime/XsAnime.kt

```diff
@@ -0,0 +1,114 @@
+package eu.kanade.tachiyomi.animeextension.ar.xsanime
+
+import eu.kanade.tachiyomi.animesource.model.AnimeFilterList
+import eu.kanade.tachiyomi.animesource.model.SAnime
+import eu.kanade.tachiyomi.animesource.model.SEpisode
+import eu.kanade.tachiyomi.animesource.model.Video
+import eu.kanade.tachiyomi.animesource.online.ParsedAnimeHttpSource
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.await
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.coroutines.runBlocking
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.lang.Exception
+
+class XsAnime : ParsedAnimeHttpSource() {
+
+    override val name = "XS Anime"
+
+    override val baseUrl = "https://ww.xsanime.com"
+
+    override val lang = "ar"
+
+    override val supportsLatest = false
+
+    override val client: OkHttpClient = network.cloudflareClient
+
+    override fun popularAnimeSelector(): String = "ul.boxes--holder div.itemtype_anime a"
+
+    override fun popularAnimeRequest(page: Int): Request = GET("$baseUrl/anime_list/page/$page")
+
+    override fun popularAnimeFromElement(element: Element): SAnime {
+        val anime = SAnime.create()
+        anime.setUrlWithoutDomain(element.attr("href"))
+        anime.title = element.attr("title")
+        anime.thumbnail_url = element.select("div.itemtype_anime_poster img").first().attr("abs:src")
+        return anime
+    }
+
+    override fun popularAnimeNextPageSelector(): String = "ul.page-numbers li a.next"
+
+    override fun episodeListSelector() = "div.EpisodesList > a"
+
+    override fun episodeFromElement(element: Element): SEpisode {
+        val episode = SEpisode.create()
+        episode.setUrlWithoutDomain(element.attr("abs:href"))
+        episode.name = element.select("a > em").text()
+
```

Comment:
you should set the episode number (you can probably extract it from a string on the website)

Reply:
probably something like `episode.episode_number = element.select("a > em").text().removePrefix("الحلقة ").toFloat()` could work...
you need to test it yourself though.

Reply:
Ok I will add it now and update the PR

Reply:
and ithink ther's no need to use `.removePrefix("الحلقة ")` as the selectors is for the episode number only it won't include ("الحلقة ") 
so it could be `episode.episode_number = element.select("a > em").text().toFloat()`
i tested it and it works

Reply:
ok, very good, again, thanks for your contributions!